### PR TITLE
Brass knuckles (now with correct branch)

### PR DESCRIPTION
**Pull request by c420-o**
## About the pull request
One day, you're minding your own business aboard the station, (beating the shit out of a talking mime, breaking into the captain's office, noogie-ing an assistant to death) and then all of a sudden you stumble upon this:
image
at first, you think _"Oh, it's a poster, nothing unusual here."_
but then you notice something. the brass knuckles.
suddenly you start questioning things. Why aren't there any brass knuckles aboard the station? why are my fists so unrobust without tator items? how will I possibly punch shit now? if the knuckles aren't real, does that mean I have no knuckles? is my body real? Am I real?
suddenly it becomes to much. Your immersion is broken. your day is ruined. not even the psychologist can save you now.
image
 but wait! there's a solution! I present to you, the brass knu-
image
oh...uh...
well she was just an assistant. wait for another famous youtuber to talk about this game and we'll get 20 fold the amount of assistants. 
## Anyways,

Introducing, the brass knuckles! finally all your immersion is fully repaired and the balance of the universe is restored. The brass knuckles are now finally here, craftable using a single bronze sheet, and now you'll be able to robust anyone and anything with just your fists. Just don't let ~Korol Konets~ someone with a sleeping carp scroll craft it.
## Changelog
:cl:
add: makes the brass knuckles
/:cl:

### CODE REVIEW

File: code/datums/martial/brassknuckles.dm [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+/datum/martial_art/brassknuckles/proc/punch(mob/living/A, mob/living/D)
+	var/obj/item/bodypart/affecting = D.get_bodypart(ran_zone(A.zone_selected))
+	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+	var/atk_verb = pick("pummels", "smokes", "wallops", "knucks", "knuckledusts")
+	D.visible_message("<span class='danger'>[A] [atk_verb]s [D]!</span>", \
+					"<span class='userdanger'>[A] [atk_verb]s you!</span>", null, null, A)
+	to_chat(A, "<span class='danger'>You [atk_verb] [D]!</span>")
+	playsound(get_turf(D), 'sound/weapons/punch1.ogg', 25, TRUE, -1)
+	log_combat(A, D, "knuckledusted (brass knuckles))")
+	D.apply_damage(20, BRUTE, affecting)
```
Comment (CRITAWAKETS) [inline code/datums/martial/brassknuckles.dm:14]: jesus christ these knuckles do more damage than a spear to the face

**Comment by TheBonded** [comment]
Where do you get bronze?

**Comment by Jared-Fogle** [comment]
i have reason to suspect you did not test this

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 2]
```diff
@@ -563,7 +563,8 @@ GLOBAL_LIST_INIT(bronze_recipes, list ( \
 	new/datum/stack_recipe("directional bronze window", /obj/structure/window/bronze/unanchored, time = 0, on_floor = TRUE, window_checks = TRUE), \
 	new/datum/stack_recipe("fulltile bronze window", /obj/structure/window/bronze/fulltile/unanchored, 2, time = 0, on_floor = TRUE, window_checks = TRUE), \
 	new/datum/stack_recipe("pinion airlock assembly", /obj/structure/door_assembly/door_assembly_bronze, 4, time = 50, one_per_turf = TRUE, on_floor = TRUE), \
-	new/datum/stack_recipe("bronze pinion airlock assembly", /obj/structure/door_assembly/door_assembly_bronze/seethru, 4, time = 50, one_per_turf = TRUE, on_floor = TRUE), \
+	new/datum/stack_recipe("bronze pinion airlock assembly", /obj/structure/door_assembly/door_assembly_bronze/seethru, 4, time = 50, one_per_turf = TRUE, on_floor = TRUE),
```
Comment (Fikou) [inline code/game/objects/items/stacks/sheets/sheet_types.dm:566]: why remove that thing at the end

**Comment by Jared-Fogle** [comment]
please test your code

File: code/datums/martial/brassknuckles.dm (hunk 1 above)
Comment (interestingusernam3) [reply code/datums/martial/brassknuckles.dm:14]: This makes sense but 20 damage makes this crit just as fast as a fire axe (24).

IMO you should lower this significantly and make it cause bone fractures easier than normal weapons (because that's what brass knuckles do).

**Comment by ilikesaying** [comment]
> Where do you get bronze?

splash copper on metal sheets with a 1u:1 ratio

File: code/game/objects/items/stacks/sheets/sheet_types.dm (hunk 2 above)
Comment (c420-o) [reply code/game/objects/items/stacks/sheets/sheet_types.dm:566]: Oversight on my part. Will readd

**Comment by c420-o** [comment]
> i have reason to suspect you did not test this
I'm not sure what you mean. Elaborate.

> please test your code. also that dmg number is ridiculous

Yeah, the damage number I can understand. But you also have to take into account that you miss a pretty hefty amount of punches in this game

File: code/datums/martial/brassknuckles.dm (hunk 1 above)
Comment (c420-o) [reply code/datums/martial/brassknuckles.dm:14]: Yeah that's true. I don't know how to cause bone fractures so I just assumed high damage did it. Thank you for the suggestions!

**Comment by c420-o** [comment]
I'd edit it on my phone but I know you guys hate webedits so I'll just wait until I'm home to change it

**Comment by c420-o** [comment]
Current things to work on
- call punch proc
- bone fracturing
- damage rebalance
- improved coding in brassknuckles.dm
- add brassknuckles.dm to tgstation.dme

**Comment by D4C-420** [comment]
put them in a burger for a knuckle sandwich.

**Comment by c420-o** [comment]
> put them in a burger for a knuckle sandwich. 

Will do

**Comment by ATH1909** [comment]
> Just don't let Korol Konets someone with a sleeping carp scroll craft it.

Because this is a martial art, it's actually incompatible with sleeping carp.

File: code/datums/martial/brassknuckles.dm [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/datum/martial_art/brassknuckles
+	name = "Brass Knuckles"
+	id = MARTIALART_BRASSKNUCKLES
+
+/datum/martial_art/brassknuckles/proc/punch(mob/living/A, mob/living/D)
```
Comment (ATH1909) [inline code/datums/martial/brassknuckles.dm:5]: nothing calls this punch() proc

**Comment by ATH1909** [comment]
> But you also have to take into account that you miss a pretty hefty amount of punches in this game, but I'll lower it. The punches still miss anyways so I might as well remove the damage code and change it to be my own thing instead of copy-pasted sleeping carp

The harm_act() code for a martial art is executed (and deals its damage and such) before a punch's miss chance is calculated, which might not happen at all if the martial art's harm_act() code returns FALSE to keep the normal punch code form executing (which is true for sleeping carp, krav maga, and multiple other martial arts).

**Comment by c420-o** [comment]
> > But you also have to take into account that you miss a pretty hefty amount of punches in this game, but I'll lower it. The punches still miss anyways so I might as well remove the damage code and change it to be my own thing instead of copy-pasted sleeping carp
> 
> The harm_act() code for a martial art is executed (and deals its damage and such) before a punch's miss chance is calculated, which might not happen at all if the martial art's harm_act() code returns FALSE to keep the normal punch code form executing (which is true for sleeping carp, krav maga, and multiple other martial arts).
> 
> In layman's terms, the punches of most martials arts don't miss (unless they're blocked).

yeah turned out I made a pretty big mistake. I'm fixing it now

**Comment by Jared-Fogle** [comment]
> I did, but I rushed it so I didn't test it as much as I should've. I'll do more extensive testing this evening.

you didn't test the *one* feature you added that isn't just new gloves?

**Comment by c420-o** [comment]
> > I did, but I rushed it so I didn't test it as much as I should've. I'll do more extensive testing this evening.
> 
> you didn't test the _one_ feature you added that isn't just new gloves?

I didn't realize that I didn't add it to the dme. I'm testing it now in the new commit that I'll add soon

**Comment by TheBonded** [comment]
> > Where do you get bronze?
> 
> splash copper on metal sheets with a 1u:1 ratio

Copper? I guess I haven't played tg in a while but I'd be surprised if I missed a whole new chemical being added

**Comment by Jared-Fogle** [comment]
copper has always existed ya dip

**Comment by TheBonded** [comment]
> copper has always existed ya dip

Am dumb, sorry.
I guess I was expecting it to be a metal that is ground into a chemical, since I don't think I've ever used it in any recipe.

**Comment by c420-o** [comment]
wait what
